dask/array/core.py

```diff
@@ -378,6 +378,10 @@ def rec_concatenate(arrays, axis=0):
         arrays = [a[None, ...] for a in arrays]
     if len(arrays) == 1:
         return arrays[0]
+    for a in arrays:
```

Comment:
This throws out arrays in the concatenation list that have shapes of all zeros. I could probably change it to throw out all arrays with `size == 0`.

The problem is that when ghosting depth is zero the ghost slice we want to add to the array is empty and has 0 depth along the concatenation axis. This raises a `ValueError` in `np.concatenate` since the shapes in this axis do not match.